test/Sema/diag_custom_equatable_syntheszied_hashable.swift

```diff
@@ -0,0 +1,49 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend -emit-module -o %t/NormalLibrary.swiftmodule %S/Inputs/external_equatable_conformance.swift
+
+// RUN: %target-typecheck-verify-swift -I %t
+
+import NormalLibrary
+
+// All conformance synthesized no warning emitted
+struct NoCustomEquatable: Hashable {
+  let a:Int
+}
+
+// Equatable implemented and Hashable synthesized raise a warning
+enum CustomEquatable: Hashable {
+  case a
+
+  static func == (lhs: CustomEquatable, rhs: CustomEquatable) -> Bool {
+    // expected-warning@-1 {{automatically generated 'Hashable' implementation for type 'CustomEquatable' may not match the behavior of custom '==' operator}}
+    // expected-note@-2 {{add a custom 'hash(into:)' method}}
+    return true
+  }
+}
+
+// All conformance implemented no warning emitted
+enum CustomHashable: Hashable {
+  case a
+
+  func hash(into hasher: inout Hasher) {}
+
+  static func == (lhs: CustomHashable, rhs: CustomHashable) -> Bool {
+    return true
+  }
+}
+
+struct ExtendedConformance: Hashable {
+  let a:Int
+}
+
+extension ExtendedConformance {
+  static func == (lhs: ExtendedConformance, rhs: ExtendedConformance) -> Bool {
+    // expected-warning@-1 {{automatically generated 'Hashable' implementation for type 'ExtendedConformance' may not match the behavior of custom '==' operator}}
+    // expected-note@-2 {{add a custom 'hash(into:)' method}}
+    return true
+  }
+}
+
+enum ImportedConformance: ImplicitEquatable {
+  case x = 1
+}
```

Comment:
(Also, typo: I think the "z" is in the wrong place in this filename.)

Reply:
Thanks for the catch on the typo in the filename! I've added your example cases for not forgetting them. I will iterate over them and add the correct variant for out of file and cross module. But I'm always stuck on the crasher that I'm getting for asking the witness declaration of the equals function when is cross module for an enum. I don't know what check I must execute to avoid the `!witnessDC->isInnermostContextGeneric()` assertion in the `AST/ProtocolConformance.cpp` file.